src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -1733,30 +1737,197 @@ public void testTotalPloidy(final int[] ploidies, final int defaultPloidy, final
+        final File tmpDir = createTempDir("createVCFTest");
+        final File outFile = new File(tmpDir.getAbsolutePath(), "createVCFTest" + ".vcf");
+
+        final VariantContextWriter vcw = GATKVariantContextUtils.createVCFWriter(outFile, null, false);
         vcw.close();
+
         final File outFileIndex = new File(outFile.getAbsolutePath() + ".idx");
+        final File outFileMD5 = new File(outFile.getAbsolutePath() + ".md5");
+
         Assert.assertTrue(outFile.exists());
         Assert.assertFalse(outFileIndex.exists());
+        Assert.assertFalse(outFileMD5.exists());
     }
 
-    @Test
-    public void testCreateVariantContextWriterWithOptions() {
-        final File outFile = createTempFile("testVCFWriter", ".vcf");
-        final VariantContextWriter vcw = GATKVariantContextUtils.createVCFWriter(
-                outFile, new SAMSequenceDictionary(), true, Options.INDEX_ON_THE_FLY, Options.ALLOW_MISSING_FIELDS_IN_HEADER);
-        vcw.close();
-        Assert.assertTrue(outFile.exists());
-        final File outFileIndex = new File(outFile.getAbsolutePath() + ".idx");
-        Assert.assertTrue(outFile.exists());
-        Assert.assertTrue(outFileIndex.exists());
+    @DataProvider(name="createVCFWriterData")
+    public Object[][] createVCFWriterData() {
+        return new Object[][]{
+                {".vcf", ".idx", true, true},
+                {".vcf", ".idx", false, true},
+                {".vcf", ".idx", true, false},
+
+                {".bcf", ".idx", true, true},
+                {".bcf", ".idx", false, true},
+                {".bcf", ".idx", true, false},
+
+                {".vcf.bgz", ".tbi", true, true},
+                // AbstractFeatureReader fails to recognize this as block compressed unless it has an accompanying index
+                // Is that correct behavior ?
+                //".vcf.bgz", ".tbi", false, true},
+                {".vcf.gz", ".tbi", false, true},
+                {".vcf.bgz", ".tbi", true, false},
+
+                // defaults to .vcf
+                {".tmp", ".idx", false, true},
+        };
+    }
+    @Test(dataProvider = "createVCFWriterData")
+    public void testCreateVCFWriterWithOptions(
+            final String outputExtension,
+            final String indexExtension,
+            final boolean createIndex,
+            final boolean createMD5) throws IOException {
+
+        final File tmpDir = createTempDir("createVCFTest");
+        final File outputFile = new File(tmpDir.getAbsolutePath(), "createVCFTest" + outputExtension);
+
+        Options options[] = createIndex ?
+                new Options[] {Options.INDEX_ON_THE_FLY} :
+                new Options[] {};
+        try (final VariantContextWriter writer = GATKVariantContextUtils.createVCFWriter(
+                outputFile,
+                makeSimpleSequenceDictionary(),
+                createMD5,
+                options)) {
+            writeHeaderAndBadVariant(writer, false);
```

Comment:
I'm not, its just a bad name. The second arg ("false" in this case) determines whether to write the bad variant. I've separated this into two separate methods to make it clearer.